pkg/handler/v2/provider/kubevirt.go

```diff
@@ -42,60 +42,106 @@ type KubeVirtGenericReq struct {
+			kubeconfig = credentials.Kubeconfig
+		}
+	}
+	return kubeconfig, nil
+}
+
 // KubeVirtVMIPresetsEndpoint handles the request to list available KubeVirtVMIPresets (provided credentials).
 func KubeVirtVMIPresetsEndpoint(presetsProvider provider.PresetProvider, userInfoGetter provider.UserInfoGetter, settingsProvider provider.SettingsProvider) endpoint.Endpoint {
 	return func(ctx context.Context, request interface{}) (interface{}, error) {
-		req := request.(KubeVirtGenericReq)
-		kubeconfig := req.Kubeconfig
+		kubeconfig, err := kubeconfigFromRequest(ctx, request, presetsProvider, userInfoGetter)
+		if err != nil {
+			return nil, err
+		}
+		return providercommon.KubeVirtVMIPresets(ctx, kubeconfig, nil, settingsProvider)
+	}
+}
 
-		userInfo, err := userInfoGetter(ctx, "")
+// KubeVirtInstancetypesEndpoint handles the request to list available KubeVirtInstancetypes (provided credentials).
+func KubeVirtInstancetypesEndpoint(presetsProvider provider.PresetProvider, userInfoGetter provider.UserInfoGetter, settingsProvider provider.SettingsProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		kubeconfig, err := kubeconfigFromRequest(ctx, request, presetsProvider, userInfoGetter)
 		if err != nil {
-			return nil, common.KubernetesErrorToHTTPError(err)
+			return nil, err
 		}
-		if len(req.Credential) > 0 {
-			preset, err := presetsProvider.GetPreset(ctx, userInfo, req.Credential)
-			if err != nil {
-				return nil, utilerrors.New(http.StatusInternalServerError, fmt.Sprintf("can not get preset %s for user %s", req.Credential, userInfo.Email))
-			}
-			if credentials := preset.Spec.Kubevirt; credentials != nil {
-				kubeconfig = credentials.Kubeconfig
-			}
+		return providercommon.KubeVirtInstancetypes(ctx, kubeconfig, nil, settingsProvider)
+	}
+}
+
+// KubeVirtPreferencesEndpoint handles the request to list available KubeVirtPreferences (provided credentials).
+func KubeVirtPreferencesEndpoint(presetsProvider provider.PresetProvider, userInfoGetter provider.UserInfoGetter, settingsProvider provider.SettingsProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		kubeconfig, err := kubeconfigFromRequest(ctx, request, presetsProvider, userInfoGetter)
+		if err != nil {
+			return nil, err
 		}
-		return providercommon.KubeVirtVMIPresets(ctx, kubeconfig, nil, settingsProvider)
+		return providercommon.KubeVirtPreferences(ctx, kubeconfig, nil, settingsProvider)
 	}
 }
 
 // KubeVirtVMIPresetsWithClusterCredentialsEndpoint handles the request to list available KubeVirtVMIPresets (cluster credentials).
 func KubeVirtVMIPresetsWithClusterCredentialsEndpoint(projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider, seedsGetter provider.SeedsGetter, userInfoGetter provider.UserInfoGetter, settingsProvider provider.SettingsProvider) endpoint.Endpoint {
```

Comment:
please mark as deprecated

Reply:
done

Reply:
sure, done. 
Note that I love the kindness of your wording: "fine-tune this a little bit" :-)  how could I refuse ?

Reply:
:-)